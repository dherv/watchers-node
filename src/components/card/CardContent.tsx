import React from "react";
import styled, { ThemeProvider } from "styled-components";

const CardContent = ({ title, theme }: { title: string; theme: any }) => (
  <ThemeProvider theme={theme}>
    <Container>
      <Title>{title}</Title>
      <ScoreContainer>
        <Metacritic>
          <span>Metacritic</span>
          <span>89</span>
        </Metacritic>
        <IMDb>
          <span>IMDb</span>
          <span>8.9</span>
        </IMDb>
      </ScoreContainer>
    </Container>
  </ThemeProvider>
);

const Container = styled.div`
  padding: 2rem 1rem;
  background-color: #323534;
  border-radius: 0 0 10px 10px;
`;

const Title = styled.h4`
  font-weight: 600;
  line-height: 1.5em;
  height: 3em;
  overflow: hidden;
  font-size: ${props => props.theme.fontSize};
`;

const ScoreContainer = styled.div`
  display: ${props => props.theme.textDisplay};
  margin-top: 1rem;
`;

const Score = styled.div`
  font-size: 12px;
  margin-right: 1rem;
  span:first-child {
    font-weight: 600;
    margin-right: 8px;
  }
`;

const Metacritic = styled(Score)``;
const IMDb = styled(Score)``;

export { Score, ScoreContainer, Metacritic, IMDb, Container, Title };

export default CardContent;
