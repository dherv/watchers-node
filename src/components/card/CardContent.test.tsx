import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import CardContent, { Title, Metacritic, IMDb } from "./CardContent";

describe("CardContent", () => {
  const props = { title: "Title", theme: {} };
  const wrapper = shallow(<CardContent {...props} />);

  describe("CardContent component", () => {
    describe("user interface", () => {
      test("should display a Title styled component with prop.title", () => {
        expect.assertions(1);
        expect(wrapper.find(Title).text()).toEqual(props.title);
      });
      describe("should display a ScoreContainer", () => {
        test("should display a Metracritic styled component", () => {
          expect.assertions(2);
          expect(wrapper.find(Metacritic)).toHaveLength(1);
          expect(
            wrapper
              .find(Metacritic)
              .find("span")
              .first()
              .text()
          ).toEqual("Metacritic");
        });
        test("should display a IMDb styled component", () => {
          expect.assertions(2);
          expect(wrapper.find(IMDb)).toHaveLength(1);
          expect(
            wrapper
              .find(IMDb)
              .find("span")
              .first()
              .text()
          ).toEqual("IMDb");
        });
      });
    });
    describe("events", () => {});
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer.create(<CardContent {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
