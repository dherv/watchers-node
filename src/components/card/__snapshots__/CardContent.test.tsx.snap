// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent snapshot should match 1`] = `
.c0 {
  padding: 2rem 1rem;
  background-color: #323534;
  border-radius: 0 0 10px 10px;
}

.c1 {
  font-weight: 600;
  line-height: 1.5em;
  height: 3em;
  overflow: hidden;
}

.c2 {
  margin-top: 1rem;
}

.c3 {
  font-size: 12px;
  margin-right: 1rem;
}

.c3 span:first-child {
  font-weight: 600;
  margin-right: 8px;
}

.c4 {
  font-size: 12px;
  margin-right: 1rem;
}

.c4 span:first-child {
  font-weight: 600;
  margin-right: 8px;
}

<div
  className="c0"
>
  <h4
    className="c1"
  >
    Title
  </h4>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <span>
        Metacritic
      </span>
      <span>
        89
      </span>
    </div>
    <div
      className="c4"
    >
      <span>
        IMDb
      </span>
      <span>
        8.9
      </span>
    </div>
  </div>
</div>
`;
